backends/graphics/windowed.h

```diff
@@ -120,8 +120,11 @@ class WindowedGraphicsManager : virtual public GraphicsManager {
 		x = CLIP<int>(x, sourceX, sourceMaxX);
 		y = CLIP<int>(y, sourceY, sourceMaxY);
 
-		return Common::Point(((x - sourceX) * targetWidth + sourceWidth / 2) / sourceWidth,
-		                     ((y - sourceY) * targetHeight + sourceHeight / 2) / sourceHeight);
+		int virtualX = ((x - sourceX) * targetWidth + sourceWidth / 2) / sourceWidth;
+		int virtualY = ((y - sourceY) * targetHeight + sourceHeight / 2) / sourceHeight;
+
+		return Common::Point(CLIP<int>(virtualX, 0, targetWidth - 1),
+				             CLIP<int>(virtualY, 0, targetHeight - 1));
```

Comment:
The indentation is a bit off die to the use of two additional tabs (after the first two tabs) instead of spaces.